lib/Conversion/DistToStandard/DistToStandard.cpp

```diff
@@ -31,23 +35,35 @@
 #include "../PassDetail.h"
 
 namespace imex {
-namespace {
+namespace dist {
 
-// create call to given function with 3 args: guid, new rank-shaped 1d tensor
-// and rank
-inline ::mlir::Value
-createCallGetRankedData(::mlir::Location &loc, ::mlir::OpBuilder &builder,
-                        const char *func, ::mlir::Value guid, uint64_t rank) {
-  auto rankV = createIndex(loc, builder, rank);
-  auto tnsr = builder.create<::mlir::tensor::EmptyOp>(
-      loc, ::mlir::ArrayRef<::mlir::OpFoldResult>({rankV}),
-      builder.getI64Type());
-  auto fsa = builder.getStringAttr(func);
-  (void)builder.create<::mlir::func::CallOp>(
-      loc, fsa, ::mlir::TypeRange(),
-      ::mlir::ValueRange({guid, tnsr, createInt(loc, builder, rank)}));
-  return tnsr;
-}
+/// After full conversion the cast is expected
+/// to have no use. FIXME Is there a better/cleaner way to do this?
+::mlir::Value materializeDistTensor(::mlir::OpBuilder &builder,
+                                    ::mlir::Location loc, ::mlir::Value gshape,
+                                    ::mlir::Value ltensor,
+                                    ::mlir::Value loffsets,
+                                    ::mlir::Value team) {
+  // Put named arguments into given Vector in a well defined order,
+  // so that extraction is correct
+  ::mlir::SmallVector<::mlir::Value> vals(::imex::dist::INFO_LAST);
```

Comment:
`std::array` as size is known at compile time.

Reply:
done.